packages/react-paypal-js/src/v6/hooks/useCardFieldsOneTimePaymentSession.test.ts

```diff
@@ -0,0 +1,221 @@
+import { act, renderHook } from "@testing-library/react-hooks";
+
+import { expectCurrentErrorValue } from "./useErrorTestUtil";
+import { useCardFieldsSession } from "./useCardFields";
+import { useCardFieldsOneTimePaymentSession } from "./useCardFieldsOneTimePaymentSession";
+import { CARD_FIELDS_SESSION_TYPES } from "../components/CardFieldsProvider";
+import { toError } from "../utils";
+
+import type {
+    CardFieldsOneTimePaymentSession,
+    OneTimePaymentFlowResponse,
+} from "../types";
+
+jest.mock("./useCardFields");
+
+const mockUseCardFieldsSession = useCardFieldsSession as jest.MockedFunction<
+    typeof useCardFieldsSession
+>;
+
+const mockOrderId = "test-order-id";
+
+const mockOptions = { billingAddress: { postalCode: "12345" } };
+
+const mockSuccessfulSubmitResponse: OneTimePaymentFlowResponse = {
+    data: {
+        orderId: mockOrderId,
+    },
+    state: "succeeded",
+};
+
+// Mock Factories
+const createMockCardFieldsOneTimePaymentSession = (
+    overrides?: Partial<CardFieldsOneTimePaymentSession>,
+): CardFieldsOneTimePaymentSession => ({
+    submit: jest.fn().mockResolvedValue(undefined),
+    on: jest.fn(),
+    update: jest.fn(),
+    createCardFieldsComponent: jest.fn(),
+    ...overrides,
+});
+
+describe("useCardFieldsOneTimePaymentSession", () => {
+    let mockCardFieldsOneTimePaymentSession: CardFieldsOneTimePaymentSession;
+
+    beforeEach(() => {
+        mockCardFieldsOneTimePaymentSession =
+            createMockCardFieldsOneTimePaymentSession({
+                submit: jest
+                    .fn()
+                    .mockResolvedValue(mockSuccessfulSubmitResponse),
+            });
+
+        mockUseCardFieldsSession.mockReturnValue({
+            cardFieldsSession: mockCardFieldsOneTimePaymentSession,
+            cardFieldsSessionType: CARD_FIELDS_SESSION_TYPES.ONE_TIME_PAYMENT,
+        });
+    });
+
+    afterEach(() => {
+        jest.clearAllMocks();
+    });
+
+    test("should return the expected object structure", () => {
```

Comment:
I've seen AI generate these types of tests frequently and I don't find them useful; apologies, again, if this is copy/pasted from another suite 😅. This test is not checking the exact structure (e.g. some kind of equality), just that these names exist in the object, and not even that the values are equal to anything.

I think we should remove this test and ensure with other tests in the suite that (1) the property exists and (2) the value is correct.